rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,429 @@
+---
+feature: doc-comment-standard
+start-date: 2023-03-27
+author: hsjobeki
+co-authors: --
+shepherd-team: @DavHau; @sternenseemann; @asymmetric
+shepherd-leader: @lassulus
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Propose a standard format specification for doc-comments.
+
+This RFC includes two concerns that define a doc-comment:
+
+- Outer format rules to allow distinction between regular comments and doc-comments
+- Inner format rules that describe the required format of a doc-comment.
+
+However, both concerns relate closely to each other; It makes sense and reduces bureaucracy to address both in a single RFC.
+
+# Definitions
+
+For this RFC, we adopt the following definitions:
+
+- **doc-comment**: A structured comment documenting the code's API.
+- **internal comment**: A free-form comment for those interested in the code's implementation.
```

Comment:
```suggestion
- **regular comment**: A free-form comment for those interested in the code's implementation.
```

Reply:
Not a blocker, but I do think it's important to get the naming right before merging, and "regular comment" avoids introducing a new term for a concept which already exists.

Reply:
I'd rather have a term with intrinsic meaning rather than something derived from the evolution of the language.

```suggestion
- **implementation comment**: A free-form comment for to help with understanding the implementation and perhaps design concerns such as requirements that aren't obvious.
```

Reply:
@roberth I mean, do we need to get into what an internal/regular/implementation document does/is for? That seems out of scope :) 

The basic idea is that it's a comment that's not meant to be displayed as part of public documentation.

Reply:
I'd update the wording to use the following. This is also consistent with the current grammar/lexer: `comment` `doc-comment`.